src/jit/optcse.cpp

```diff
@@ -880,39 +972,103 @@ void Compiler::optValnumCSE_InitDataFlow()
  */
 class CSE_DataFlow
 {
-    BitVecTraits* m_pBitVecTraits;
-    EXPSET_TP     m_preMergeOut;
+    Compiler* m_comp;
+    EXPSET_TP m_preMergeOut;
+    EXPSET_TP m_callKills;
 
 public:
-    CSE_DataFlow(Compiler* pCompiler) : m_pBitVecTraits(pCompiler->cseTraits), m_preMergeOut(BitVecOps::UninitVal())
+    CSE_DataFlow(Compiler* pCompiler)
+        : m_comp(pCompiler), m_preMergeOut(BitVecOps::UninitVal()), m_callKills(BitVecOps::UninitVal())
     {
+        m_callKills = BitVecOps::MakeEmpty(m_comp->cseLivenessTraits);
+        for (unsigned inx = 0; inx < m_comp->optCSECandidateCount; inx++)
+        {
+            unsigned CseAvailBit = inx * 2;
+
+            // a one preserves availability and a zero kills the availability
+            // we generate this kind of bit pattern:  101010101010
+            //
+            BitVecOps::AddElemD(m_comp->cseLivenessTraits, m_callKills, CseAvailBit);
+        }
     }
 
     // At the start of the merge function of the dataflow equations, initialize premerge state (to detect changes.)
     void StartMerge(BasicBlock* block)
     {
-        BitVecOps::Assign(m_pBitVecTraits, m_preMergeOut, block->bbCseOut);
+        BitVecOps::Assign(m_comp->cseLivenessTraits, m_preMergeOut, block->bbCseOut);
```

Comment:
This initializes `m_preMergeOut` to `cseFull`, which is all bits set,  because we previous set `bbCseOut` to `cseFull` in `optValnumCSE_InitDataFlow()`

```
cseFull           = BitVecOps::MakeFull(cseLivenessTraits);
block->bbCseOut = BitVecOps::MakeCopy(cseLivenessTraits, cseFull);
```